commit 59e3953ce578ed82369aa6eb4f9d60c137157564 (refs/changes/04/2604/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-09-19T19:11:42+02:00 (2 years, 1 month ago)
    
    CAPI-535 runtests is incorrectly reporting success when test suite exits with non zero value

diff --git a/test/runtests b/test/runtests
index 23e3359..8b02782 100755
--- a/test/runtests
+++ b/test/runtests
@@ -20,8 +20,8 @@ if [ "$TRACE" != "" ]; then
 fi
 # With this stuff enabled we'll get not summary at the end of the tests
 # execution if any test wrote to stderr.
-#set -o errexit
-#set -o pipefail
+set -o errexit
+set -o pipefail
 
 
 
@@ -112,10 +112,13 @@ if [[ -n "$opt_test_pattern" ]]; then
     echo "# Running filtered set of test files: $test_files"
 fi
 
+set +o errexit
 if [[ -n "$test_files" ]]; then
     PATH=$NODE_INSTALL/bin:$PATH TEST_CONFIG_FILE=$TOP/etc/config.json node test/test.js \
         | tee $OUTPUT_DIR/ufds.tap
 fi
+rv=$?
+set -o errexit
 
 echo ""
 echo "# test output:"
@@ -147,3 +150,4 @@ echo ""
 if [[ ${tests} != ${passed} ]]; then
     exit 1
 fi
+exit $rv
